src/care/LoopFuser.h

```diff
@@ -1066,120 +973,137 @@ void LoopFuser::registerFree(care::host_device_ptr<T> & array) {
+// Note that __fusible_scan_pos_*_ are raw pointers being captured into a lambda,
+// In many situations this can lead to dereferencing of a host pointer, so it is desirable
+// to have a clang-query that can check for this. Below is a query that checks for raw
+// pointer captures and also ignores these particular variables.
+//
+//let comment "### Identify device lambda contexts"
+//let b1 callExpr(hasArgument(0,hasType(asString("struct care::gpu"))))
+//let b2 callExpr(hasArgument(0,hasType(asString("struct care::parallel"))))
+//let b3 cxxMemberCallExpr(on(hasType(pointsTo(cxxRecordDecl(hasName("LoopFuser"))))))
+//let rajaDeviceContext callExpr(anyOf(b1,b2,b3))
+//let inDeviceLambda hasAncestor(lambdaExpr(hasAncestor(rajaDeviceContext)))
+//let comment "### Ignore implicit casts (to prevent duplicate matches)"
+//let notInImplicitCast unless(hasAncestor(implicitCastExpr()))
+//let comment "### Set up detection of reference of raw pointer in device lambda"
+//let comment "### ignoringImplicit and no implicitCastExpr ancestor is required to prevent duplicate matches"
+//match expr(inDeviceLambda, notInImplicitCast, ignoringImplicit(declRefExpr(to(varDecl(hasType(isAnyPointer()), 
+//           unless(matchesName("__fusible_scan_pos.*__")), unless(inDeviceLambda)))))).bind("capture_of_raw_pointer_in_lambda")
+
 #define FUSIBLE_BOOKKEEPING(FUSER,START,END) \
-   auto __fusible_offset__ = FUSER->getOffset(); \
-   auto __fusible_start_index__ = START; \
-   auto __fusible_end_index__ = END;
+   FUSIBLE_KERNEL_BOOKKEEPING(FUSER) ; \
+   auto __fusible_action_index__ = FUSER->actionCount(); \
+   index_type *__fusible_scan_pos_starts__ = FUSER->getScanPosStarts(); \
+   index_type *__fusible_scan_pos_outputs__ = FUSER->getScanPosOutputs(); \
+   __fusible_start_index__ = START; \
+   auto __fusible_end_index__ = END; \
+   auto __fusible_verbose__ = LoopFuser::verbose; \
+   __fusible_offset__ = __fusible_offset__; \
+   __fusible_action_index__ = __fusible_action_index__ ; \
+   __fusible_scan_pos_starts__ = __fusible_scan_pos_starts__ ;  \
+   __fusible_scan_pos_outputs__ = __fusible_scan_pos_outputs__ ; \
+   __fusible_verbose__ = __fusible_verbose__ ;
+   
 
 // adjusts the index by adding the loop start index and subtracting off the
 // loop fuser offset to bring the loop
 // from the fuser global index space back into its own index space.
-#define FUSIBLE_INDEX_ADJUST(INDEX) INDEX += __fusible_start_index__ - __fusible_offset__ ;
+// NOTE: no adjustment necessary with 2D launch
+ #define FUSIBLE_INDEX_ADJUST(INDEX) \
+    int __fusible_global_index__ = __fusible_offset__+ INDEX; \
+    INDEX += __fusible_start_index__;  \
+    __fusible_global_index__ = __fusible_global_index__; // quiet compiler where unused
+// NOTE: adjustment necessary with 1D launch
+// #define FUSIBLE_INDEX_ADJUST(INDEX) int __fusible_global_index__ = INDEX; INDEX += __fusible_start_index__ - __fusible_offset__ ;
```

Comment:
Should this be removed?

Reply:
I'd like to leave it in in case WorkGroups ever supports 1D launches in the future.